pkg/services/sqlstore/datasource.go

```diff
@@ -136,6 +138,33 @@ func (ss *SQLStore) DeleteDataSource(ctx context.Context, cmd *datasources.Delet
 	})
 }
 
+// DeleteDataSourceSecrets removes a datasource secure_json_data by org_id as well as either uid (preferred), id, or name.
+func (ss *SQLStore) DeleteDataSourceSecrets(ctx context.Context, cmd *datasources.DeleteDataSourceSecretsCommand) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
+		if cmd.OrgID == 0 || len(cmd.UID) == 0 {
+			return datasources.ErrDataSourceIdentifierNotSet
+		}
+
+		result, err := sess.Exec("UPDATE data_source SET secure_json_data=? WHERE org_id=? AND uid=?", "{}", cmd.OrgID, cmd.UID)
+		if err != nil {
+			return err
+		}
+
+		cmd.DeletedSecretsCount, _ = result.RowsAffected()
+
+		// Publish data source deletion event
+		sess.publishAfterCommit(&events.DataSourceSecretDeleted{
```

Comment:
Are you using this for something? If not, can skip it?

Reply:
Will remove, I thought it was the usual to publish after commit for every SQL action.

Reply:
This will be completely removed with the method in favor of changes suggested on https://github.com/grafana/grafana/pull/50463#discussion_r913141904